pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -122,52 +123,58 @@ func (r *DNSPolicyReconciler) expectedHealthCheckProbesForGateway(ctx context.Co
 		interval = *dnsPolicy.Spec.HealthCheck.Interval
 	}
 
-	for _, address := range gw.Status.Addresses {
-		matches := strings.Split(address.Value, "/")
-		if len(matches) != 2 {
-			log.V(3).Info(fmt.Sprintf("%s cannot be processed. expected <clsutername>/<hostname/ipaddress>", address.Value))
-			return nil
+	gatewayWrapper := utils.NewGatewayWrapper(gw.Gateway)
+	if err := gatewayWrapper.Validate(); err != nil {
+		return nil
+	}
+
+	clusterGatewayAddresses := gatewayWrapper.GetClusterGatewayAddresses()
+
+	for _, listener := range gw.Spec.Listeners {
+
+		//skip wildcard listeners
+		if strings.Contains(string(*listener.Hostname), "*") {
+			continue
 		}
 
-		for _, listener := range gw.Spec.Listeners {
-			if strings.Contains(string(*listener.Hostname), "*") {
-				continue
-			}
+		port := dnsPolicy.Spec.HealthCheck.Port
+		if port == nil {
+			listenerPort := int(listener.Port)
+			port = &listenerPort
+		}
 
-			port := dnsPolicy.Spec.HealthCheck.Port
-			if port == nil {
-				listenerPort := int(listener.Port)
-				port = &listenerPort
-			}
+		// handle protocol being nil
+		var protocol string
```

Comment:
could poss drop the else? 
```
protocol := dnsPolicy.Spec.HealthCheck.Protocol 
if protocol == nil{
protocol = string(listener.Protocol)
}
```

Reply:
Can't do it quite like that, but I've changed it so there's no if now.

Reply:
oh no, can't do that either. I've set it back to the original - this isn't new code, I've just moved it.